src/getJavaOptions.ts

```diff
@@ -15,11 +15,18 @@ function jvmOpts(outputChannel: OutputChannel): string[] {
 
 function javaOpts(outputChannel: OutputChannel): string[] {
   const javaOpts = process.env.JAVA_OPTS;
+  const options = [];
   if (javaOpts) {
     outputChannel.appendLine("Using JAVA options set in JAVA_OPTS");
-    return parse(javaOpts);
+    for (const option of javaOpts) {
+      if (typeof option === 'string') {
+        options.push(option)
```

Comment:
I'm probably being extra picky, but I don't really like loops and mutability (even in JS :D)

For consistency with the rest of the code, you would mind refactoring this to use `.reduce` or other combinators?

Reply:
I don't either, and I tried, but combinators don't work well with type checks, i.e. `.filter(typeof _ === string)` doesn't propagate the type constraint to the result.

I'll try again, and harder :blush:

Reply:
yeah, filter is particularly stupid about it.
You can use a custom guard to make it smarter I think

```
function isString(x: unknown): x is string {
  return typeof x === 'string'
}
```

or using `reduce` (which you might need anyway to log the errors) should fix it

Reply:
@swiesner-dlr Thanks.  My typescript foo isn't the best :blush: